packages/data-sql/src/query-converter/converter.ts

```diff
@@ -46,5 +48,8 @@ export const convertToAbstractSqlQueryAndGenerateAliases = (abstractQuery: Abstr
 		statement.order = convertSort(abstractQuery.modifiers.sort);
 	}
 
-	return statement;
+	return {
+		sql: statement,
+		aliasMapping: convertedSqlAndAliasMap.aliasMapping,
+	};
```

Comment:
Should we move `parameters` up to this level as well?

Reply:
yes can do. since this bit of ORM logic is now tightly coupled to the actual query anyways due to the random alias, we could group query, parameters and alias map here as well. and separating the parameters a bit from the query clauses seems to be a good idea

Reply:
alternatively I'm thinking about leave it as is for now or separating the parameters within the `AbstractSqlQuery` so this current type 
```ts
interface AbstractSqlQuery {
	select: (AbstractSqlQuerySelectNode | AbstractSqlQueryFnNode)[];
	from: string;
        ...
	parameters: ParameterTypes[];
}
```
will get to something like this:
```ts
interface AbstractSqlQuery {
       clauses: {
	     select: (AbstractSqlQuerySelectNode | AbstractSqlQueryFnNode)[];
	     from: string;
             ...
       },
       parameters: ParameterTypes[];
}
```
this could be handy when we work on the other relational types soon and work on the question how a driver knows when to use a sub query or a join. we probably need to store some abstract nodes in there instead of those clauses, with which the driver can use to make either a sub query or to a join out of them. so we may need to add those "flexible" nodes in the `abstractSqlQuery` soon like this:
```ts
interface AbstractSqlQuery {
       fixedClauses: { // clauses that each and every driver will use
	     select: (AbstractSqlQuerySelectNode | AbstractSqlQueryFnNode)[];
	     from: string;
             ...
       },
       abstractNodes: { ... } // like relational nodes which the driver needs to convert them to specific clauses later on
       parameters: ParameterTypes[];
}
```
this is just a first thought about how to realize this if we really want the driver to determine, which construct to use.
 
since moving the parameters out of the type is noticeable amount of work, I'm not sure if we should optimize this at this point or rather wait until we've figured out this other stuff regarding relations first. I'm hoping to reduce back and forth work this way.

Reply:
> Should we move parameters up to this level as well?

it's now on the same level. I called the select, where nodes etc. `clauses`. 

https://github.com/directus/directus/blob/9846cf2156be82629df6e36af2a5b021d791852c/packages/data-sql/src/query-converter/converter.ts#L36-L40

therefore I changed the `abstractSqlQuery` type to this https://github.com/directus/directus/blob/9846cf2156be82629df6e36af2a5b021d791852c/packages/data-sql/src/types/abstract-sql-query.ts#L23-L27

Reply:
Perfect 🚀